src/components/wallet/earn/UndepositButtons.vue

```diff
@@ -0,0 +1,305 @@
+
+@Component({
+    components: {
+        ModalUndeposit,
+        CamBtn,
+        ModalAbortSigning,
+        Alert,
+    },
+})
+export default class UndepositButtons extends Vue {
+    @Prop() pendingUndepositTx!: UndepositPendingTx
+    @Prop() reward!: PlatformRewardDeposit
+
+    // @ts-ignore
+    helpers = this.globalHelper()
+
+    buttonStatus: number = 0
+    initMultisig: boolean = false
+
+    $refs!: {
+        abort: ModalAbortSigning
+        modal_undeposit: ModalUndeposit
+    }
+
+    @Watch('pendingTX')
+    watchPendingTX() {
+        this.updateButtonStatus()
+    }
+
+    mounted() {
+        this.updateButtonStatus()
+    }
+
+    initMultisigTx() {
+        this.$emit('updateDisclaimer')
+        this.initMultisig = true
+    }
+
+    cancelInitMultisigTx() {
+        this.$emit('updateDisclaimer')
+        this.initMultisig = false
+    }
+
+    get pendingTX(): SignavaultTx | undefined {
+        let pendingUndepositTx = this.pendingUndepositTx?.pendingTx as SignavaultTx
+        return pendingUndepositTx
+    }
+
+    get unlockableAmount(): BN {
+        return this.reward.deposit.unlockableAmount
+    }
+
+    get isUndepositDisabled() {
+        return this.reward.deposit.unlockableAmount.isZero()
+    }
+
+    get activeWallet() {
+        return this.$store.state.activeWallet
+    }
+
+    get sigValue() {
+        return this.pendingTX?.tx.owners?.filter((owner) => !!owner.signature)?.length
+    }
+
+    get threshold() {
+        return this.pendingTX?.tx.threshold
+    }
+
+    get signStatus(): boolean {
+        let isSigned = false
+        this.txOwners.forEach((owner) => {
+            if (
+                this.activeWallet.wallets.find((w) => w?.getAllAddressesP()?.[0] === owner.address)
+            ) {
+                if (owner.signature) isSigned = true
+            }
+        })
+        return isSigned
+    }
+
+    get canExecuteMultisigTx(): boolean {
+        let signers = 0
+        let threshold = this.pendingTX?.tx?.threshold
+        this.txOwners.forEach((owner) => {
+            if (owner.signature) signers++
+        })
+        if (threshold) return signers >= threshold
+        return false
+    }
+
+    get disableSignButton(): boolean {
+        let isSigned = false
+        this.txOwners.forEach((owner) => {
+            if (
+                this.activeWallet.wallets.find((w) => w?.getAllAddressesP()?.[0] === owner.address)
+            ) {
+                if (owner.signature) isSigned = true
+            }
+        })
+        return isSigned
+    }
+
+    get signavaultPendingTx() {
+        return this.$store.getters['Signavault/transactions'].length > 0
+    }
+
+    get txOwners() {
+        return this.pendingTX?.tx?.owners ?? []
+    }
+
+    updateButtonStatus() {
+        if (
+            this.pendingUndepositTx?.depositTxIDs?.find(
+                (tx: string) => tx === this.reward.deposit.depositTxID
+            )
+        ) {
+            this.buttonStatus = 1
```

Comment:
We should avoid using magic numbers in favor of constants to give them a recognizable name.